Components/InputTypeRadio/index.js

```diff
@@ -0,0 +1,39 @@
+import styled from "styled-components";
+
+export default function InputTypeRadio({
+  nameString,
+  valueString,
+  onChange,
+  idString,
+  labelString,
+}) {
+  function checked() {
+    nameString === valueString;
+  }
+
+  return (
+    <>
+      <RadioButton>
+        <input
+          type="radio"
+          name={nameString}
+          value={valueString}
+          id={idString}
+          onChange={onChange}
+          checked={checked()}
```

Comment:
I like the idea of breaking logic up into smaller functions, but in the case i think it is more readable to just do the check in the jsx. If the check was to get more complex at some point then i would consider breaking the logic out to a function. 

```suggestion
          checked={nameString === valueString}
```

Reply:
This will not work because the checked attribute is a Boolean and doesn't care for its value. You have to render it conditionally.

{condition && "checked"}

Reply:
The idea behind my suggestion is that 

`nameString === valueString`

will yield a boolean value. i.e.

`"string1" === "string2" // false`
`"string1" === "string1" // true`

which means we would get

``` js
checked={false} // or
checked={true}
```

As far as i am aware, this is the way to do it in jsx.

Reply:
Aaah! It is working because it is jsx, I was so confused by this.
Thanks!

Reply:
`checked={nameString === valueString}` does indeed not work -> the user can not check the radio button then. For our project we not only need a boolean as a value for our variables. We also need other values like "reversed"  etc.. Therefore we want to keep the code we have. We have tested it further and it works for our Plotting component.